pkg/controller/infinispan/resources/secret/secret_controller.go

```diff
@@ -76,25 +73,47 @@ func Add(mgr manager.Manager) error {
 }
 
 func (s *secretResource) Process() (reconcile.Result, error) {
-	secret, err := s.getSecret(s.infinispan.GetAdminSecretName())
-	if err != nil {
-		return reconcile.Result{}, err
+	adminSecret := &corev1.Secret{
```

Comment:
It would be cleaner if we add the admin secret logic to a function that's called within process so that it's self contained.

Reply:
Yes, correct, will create a separate function. It will be cleaner solution